### PR TITLE
refs #27339: Support IPv6 primary connection

### PR DESCRIPTION
**Pull request by fabienph**

### CODE REVIEW

File: src/utils/ip.hpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <sys/types.h>
+
+struct sockaddr;
+
+[[nodiscard]]
+bool is_ipv4_mapped_ipv6(const char *) noexcept;
```
Comment (jonathanpoelen) [inline src/utils/ip.hpp:8]: All parameters should be named.

File: src/core/listen.hpp [hunk 2]
```diff
@@ -48,56 +49,11 @@
...
-    return unique_sck;
-}
-
-inline unique_fd create_server_bind_sck(
-    unique_fd fd_sck, sockaddr& addr, EnableTransparentMode enable_transparent_mode)
+inline unique_fd create_server_bind_sck
+(unique_fd fd_sck,
+ sockaddr_storage& addr,
+ socklen_t addrlen,
+ EnableTransparentMode enable_transparent_mode) noexcept
```
Comment (jonathanpoelen) [inline src/core/listen.hpp:56]: style: :vomiting_face:

File: src/core/listen.hpp [hunk 3]
```diff
@@ -159,11 +116,16 @@ inline unique_fd create_unix_server(
...
+                                  sizeof(u.s),
+                                  enable_transparent_mode);
 }
 
-inline unique_fd create_server(
-    uint32_t s_addr, int port, EnableTransparentMode enable_transparent_mode)
+inline unique_fd create_server
+(uint32_t s_addr,
+ int port,
+ EnableTransparentMode enable_transparent_mode) noexcept
```
Comment (jonathanpoelen) [inline src/core/listen.hpp:128]: style: :vomiting_face: 

etc

File: src/core/listen.hpp (hunk 2 above)
Comment (fabienph) [reply src/core/listen.hpp:56]: It's for avoid to exceed 80 columns for readability but I understand well your disagreement :/

Comment (hardening) [reply src/core/listen.hpp]: Looking forward for the code style guide ;)

**Pull request merged**